src/test/java/teammates/test/cases/webapi/DeleteFeedbackSessionActionTest.java

```diff
@@ -23,13 +28,83 @@ protected String getRequestMethod() {
     @Test
     @Override
     protected void testExecute() throws Exception {
-        // TODO
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("Typical case: Delete from recycle bin");
+
+        CourseAttributes course = typicalBundle.courses.get("typicalCourse1");
+        FeedbackSessionAttributes session = typicalBundle.feedbackSessions.get("session1InCourse1");
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, session.getFeedbackSessionName(),
+        };
+
+        assertNotNull(logic.getFeedbackSessionDetails(session.getFeedbackSessionName(), course.getId()));
+
+        logic.moveFeedbackSessionToRecycleBin(session.getFeedbackSessionName(), course.getId());
+
+        DeleteFeedbackSessionAction deleteFeedbackSessionAction = getAction(params);
+        JsonResult result = getJsonResult(deleteFeedbackSessionAction);
+        MessageOutput messageOutput = (MessageOutput) result.getOutput();
+
+        assertEquals(messageOutput.getMessage(), "The feedback session is deleted.");
+        assertNull(logic.getFeedbackSessionFromRecycleBin(session.getFeedbackSessionName(), course.getId()));
+
+        ______TS("Delete session that has already been deleted");
+
+        // Will fail silently and not throw any exception
+        getJsonResult(deleteFeedbackSessionAction);
+
+        ______TS("Delete session that does not exist");
+
+        params = new String[] {
+                Const.ParamsNames.COURSE_ID, course.getId(),
+                Const.ParamsNames.FEEDBACK_SESSION_NAME, "randomName",
+        };
+
+        deleteFeedbackSessionAction = getAction(params);
```

Comment:
For these tests, you should ensure that the preconditions are met as well. For example, you should try to get feedback session "randomName" and ensure it returns null.